traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/widgets/CarouselView.java

```diff
@@ -0,0 +1,55 @@
+package com.guestlogix.traveleruikit.widgets;
+
+import android.content.Context;
+import android.util.AttributeSet;
+import android.widget.RelativeLayout;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.LinearSnapHelper;
+import androidx.recyclerview.widget.RecyclerView;
+import androidx.recyclerview.widget.SnapHelper;
+import com.guestlogix.traveleruikit.R;
+import com.guestlogix.traveleruikit.adapters.ImageCarouselRecyclerViewAdapter;
+
+import java.net.URL;
+import java.util.List;
+
+public class CarouselView extends RelativeLayout {
+    private RecyclerView recyclerView;
+    private ImageCarouselRecyclerViewAdapter adapter = new ImageCarouselRecyclerViewAdapter();
+
+    public CarouselView(Context context, AttributeSet attrs, int defStyle) {
+        super(context, attrs, defStyle);
+        initView();
+    }
+
+    public CarouselView(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        initView();
+    }
+
+    public CarouselView(Context context) {
+        super(context);
+        initView();
+    }
+
+
+    public void setImageURLList(List<URL> imageUrls) {
+        adapter.setData(imageUrls);
```

Comment:
this type of proxying should be avoided.

you should only set the adapter directly?

Reply:
updated - please re-review